pkg/controller/machine.go

```diff
@@ -358,19 +358,24 @@ func (c *Controller) syncHandler(key string) error {
 	}
 
 	// case 3.2: creates an instance if there is no node associated with the given machine
-	node, nodeExistsAndIsReady, err := c.doesNodeForMachineExistAndIsReady(machine)
-	if err != nil {
-		return fmt.Errorf("failed to check if node for machine exists ans is ready: '%s'", err)
-	}
-	if !nodeExistsAndIsReady {
+	if machine.Status.NodeRef == nil {
 		glog.V(6).Infof("Requesting instance for machine '%s' from cloudprovider because no associated node with status ready found...", machine.Name)
-		err = c.ensureInstanceExistsForMachine(prov, machine, userdataProvider, providerConfig)
-		if err != nil {
+		return c.ensureInstanceExistsForMachine(prov, machine, userdataProvider, providerConfig)
+	}
+
+	node, err := c.getNodeByNodeRef(machine.Status.NodeRef)
+	if err != nil {
+		//In case we cannot find a node for the NodeRef we must remove the NodeRef & recreate an instance on the next sync
+		if kerrors.IsNotFound(err) {
```

Comment:
Actually I am not sure if this will do it, because if the instance at the cloud provider gets deleted manually, the node will always exist, but its state will not be ready. Kube does afaik not remove nodes just because they stop posting their status

Reply:
I think what we need is `if nodes does not exist OR node status != ready { ensureInstanceExistsForMachine }`

Reply:
Should be done in https://github.com/kubermatic/machine-controller/pull/253/commits/af28abac5176315327bf1fd69ebaedc45e8af08e